internal/forms/forms.go

```diff
@@ -325,80 +326,68 @@ func (m *Manager) downloadAndUnzipForms(downloadLink string) error {
 	if err != nil {
 		return fmt.Errorf("can't download update ZIP: %w", err)
 	}
-	filename := "Standard_Forms.zip"
-	dispo := resp.Header.Get("Content-Disposition")
-	if dispo != "" && strings.Contains(dispo, "filename=") {
-		fileRe := regexp.MustCompile(`filename="(.*)"`)
-		filename = fileRe.FindStringSubmatch(dispo)[1]
-	}
-	dir := os.TempDir()
-	defer os.RemoveAll(dir)
-	zipBytes, _ := io.ReadAll(resp.Body)
 	defer resp.Body.Close()
-	zipFilePath := path.Join(dir, filename)
-	err = os.WriteFile(zipFilePath, zipBytes, 0o600)
+	f, err := ioutil.TempFile(os.TempDir(), "pat")
 	if err != nil {
+		return fmt.Errorf("can't create temp file for download: %w", err)
+	}
+	defer f.Close()
+	defer os.Remove(f.Name())
+	if _, err := io.Copy(f, resp.Body); err != nil {
 		return fmt.Errorf("can't write update ZIP: %w", err)
 	}
 
 	unzipDir := m.config.FormsPath
-	err = unzip(zipFilePath, unzipDir)
-	if err != nil {
+	if err := unzip(f.Name(), unzipDir); err != nil {
 		return fmt.Errorf("can't unzip forms update: %w", err)
 	}
 	return nil
 }
 
-func unzip(src, dest string) error {
-	// https://stackoverflow.com/a/24792688/587091
-	r, err := zip.OpenReader(src)
-	if err != nil {
-		return err
-	}
-	defer r.Close()
-
-	_ = os.MkdirAll(dest, 0o755)
-
+func unzip(srcArchivePath, dstRoot string) error {
 	// Closure to address file descriptors issue with all the deferred .Close() methods
-	extractAndWriteFile := func(f *zip.File) error {
-		rc, err := f.Open()
-		if err != nil {
-			return err
+	extractAndWriteFile := func(zf *zip.File) error {
+		if zf.FileInfo().IsDir() {
```

Comment:
Clever way to get away from branching below. Technically this is no longer a perfect reproduction of the zip file, it won't include empty directories, but that's not really behavior that we need for our case.